pkgs/README.md

```diff
@@ -30,12 +30,23 @@ Before adding a new package, please consider the following questions:
 
 * Is the package ready for general use? We don't want to include projects that are too immature or are going to be abandoned immediately. In case of doubt, check with upstream.
 * Does the project have a clear license statement? Remember that software is unfree by default (all rights reserved), and merely providing access to the source code does not imply its redistribution. In case of doubt, ask upstream.
-* How realistic is it that it will be used by other people? It's good that nixpkgs caters to various niches, but if it's a niche of 5 people it's probably too small.
+* How realistic is it that it will be used by other people? It's good that nixpkgs caters to various niches, but if it's a niche of 5 people it's probably too small. A good estimate is checking upstream issues and pull requests, or other software repositories.
+* Is the software maintained upstream? Especially security-critical components should see regular maintenance.
 * Are you willing to maintain the package? You should care enough about the package to be willing to keep it up and running for at least one complete Nixpkgs' release life-cycle.
   * In case you are not able to maintain the package you wrote, you can seek someone to fill that role, effectively adopting the package.
 
 If any of these questions' answer is no, then you should probably not add the package.
 
+Special care has to be taken with security-critical software components.
+* Any package that immediately would need to be tagged with `meta.knownVulnerabilities` is not fit for nixpkgs
+* Any package depending on a known-vulnerable library should be considered carefully.
+* Web engines like chromium and gecko receive frequent updates. Any package which does not keep up-to-date web engines should be considered vulnerable.
+* (Web-)Services working on untrusted input should have a maintained and responsible upstream.
+* Applications in the unix authentication stack such as PAM/DBUS-modules or SUID binaries should be considered carefully, and should have a maintained/responsible upstream.
+* Encryption libraries should have a maintained and responsible upstream.
```

Comment:
Any ideas how to word that most of the time we want some kind of known-competent upstream for such things?

Reply:
"known-competent" is a difficult consideration. I wrote "responsible" and considered "reasonable due diligence", but getting this concise yet precise is hard. My intention is not to outright ban niche upstream devs. I'd already be happy if upstream has available communication channels to disclose (security) issues and then acts on them.

Reply:
I pushed an attempt to specify this a bit more, hope this helps